### ISSUE TITLE
socket never close?

### ISSUE THREAD

**Issue by saknarak** [jvirtanen/node-soupbintcp#8]
when `client.logout`, `client.end`, socket `end` event never emit
I have to clear interval to end the connection
or may be the server problem?

```
    client.logout(() => {
      console.log('logout ok')
      // workaround: clear keepAlive interval
      clearInterval(client._connection?._keepAlive)
    })
```

**Comment by jvirtanen** [comment]
Hi @saknarak! 👋🏼 A good find, I need to investigate this a bit.

**Comment by gwn** [comment]
Hi, quoting from the [NASDAQ SoupbinTCP 4.0 spec](https://www.nasdaq.com/docs/SoupBinTCP%204.0.pdf):

> 2.3.4 Logout Request Packet
> The client may send a Logout Request Packet to request the connection be terminated. Upon receiving a Logout Request Packet, the server will immediately terminate the connection and close the associated TCP/IP socket.

Client behavior is unspecified therefore I think it's not a mistake to keep the connection alive. Server is supposed to close the connection after logging the user out.

Also calling `client.end()` instead of clearing the private interval looks like a better idea to me, in case the server is not complying with the spec:

```
client.logout(() => {
    console.log('logout ok')
    client.end()
})
```

I don't think this is an issue of this library.

**Comment by jvirtanen** [comment]
I agree with @gwn's comments above. Note that he also found and fixed a related issue in #14.
